giraph-core/src/main/java/org/apache/giraph/job/CombinedWorkerProgress.java

```diff
@@ -204,4 +215,19 @@ public String toString() {
     }
     return sb.toString();
   }
+
+  /**
+   * Check if this instance made progress from another instance
+   *
+   * @param lastProgress Instance to compare with
+   * @return True iff progress was made
+   */
+  public boolean madeProgressFrom(CombinedWorkerProgress lastProgress) {
```

Comment:
That's why I separated getProgressString from toString, to only contain actual progress. For different supersteps we are looking at different numbers so this seemed the easiest to compare instead of having all the if-s.